src/cxy_cache.erl

```diff
@@ -423,27 +425,35 @@ refresh_item(Type, Cache_Name, New_Gen_Id, Old_Gen_Id, Mod, Key, Object) ->
     try ets:lookup(Old_Gen_Id, Key) of
 
         %% Create a new Mod:create_key_value value if not in old generation...
-        [] -> refresh_now(Type, Cache_Name, New_Gen_Id, Mod, Key, Object);
+        [] -> case Type of
+                  key -> refresh_now(Type, Cache_Name, New_Gen_Id, Mod, Key, Object);
+                  obj -> case refresh_now(Type, Cache_Name, New_Gen_Id, Mod, Key, Object) of
+                             no_value_available    -> no_value_available;
+                             {_Version, Raw_Value} -> Raw_Value
+                         end
+              end;
 
         %% Otherwise, migrate the old value to the new generation...
         [#cxy_cache_value{key=Key} = Old_Obj] ->
-            insert_to_new_gen(Cache_Name, New_Gen_Id, Mod, Old_Obj),
+            _ = insert_to_new_gen(Cache_Name, New_Gen_Id, Mod, Old_Obj),
             %% Then try to clobber it with a newly created value.
-            Cached_Value = case {Type, Object} of
-                               {key, _} ->
-                                   create_new_value(Cache_Name, New_Gen_Id, Mod, Key);
-                               {obj, no_value_available} ->
-                                   no_value_available;
-                               {obj, _} ->
-                                   {Possibly_New_Vsn, Possibly_New_Value} = Object,
-                                   insert_value_if_newer(Cache_Name, New_Gen_Id, Mod,
-                                                         Key, Possibly_New_Vsn, Possibly_New_Value)
-                           end,
+            Cached_Value
+                = case {Type, Object} of
+                      {key, _} ->
+                          create_new_value(Cache_Name, New_Gen_Id, Mod, Key);
+                      {obj, no_value_available} ->
+                          no_value_available;
+                      {obj, _} ->
+                          {Possibly_New_Vsn, Possibly_New_Value} = Object,
+                          insert_value_if_newer(Cache_Name, New_Gen_Id, Mod,
+                                                Key, Possibly_New_Vsn, Possibly_New_Value)
+                  end,
             case Cached_Value of
-                no_value_available -> ?WHEN_GEN_EXISTS(New_Gen_Id, ets:delete(New_Gen_Id, Key)),
-                                      ?WHEN_GEN_EXISTS(Old_Gen_Id, ets:delete(Old_Gen_Id, Key)),
-                                      return_cache_miss(Cache_Name, no_value_available);
-                {_Version, _Value} -> return_cache_refresh(Cache_Name, Cached_Value)
+                {_Version, Raw_Value} -> return_cache_refresh(Cache_Name, Raw_Value);
+                no_value_available    ->
+                    ?WHEN_GEN_EXISTS(New_Gen_Id, ets:delete(New_Gen_Id, Key)),
+                    ?WHEN_GEN_EXISTS(Old_Gen_Id, ets:delete(Old_Gen_Id, Key)),
+                    return_cache_miss(Cache_Name, no_value_available)
```

Comment:
The actual error occurred here when Cached_Value was a binary rather than a tuple.